db/core.py

```diff
@@ -117,6 +123,106 @@ def _coerce_primary_key_map(cls, identity: Any) -> dict[str, Any]:
         raise ValueError("Identity must be scalar, tuple/list, or dict matching the primary key.")
 
 
+class AuditLogModel(BaseModel):
```

Comment:
Can this work as a mixin so that you can define new log models by just inheriting from the original model and this? e.g.

```
class GroupMembershipLog(GroupMembership, AuditLogMixin):
    pass
```